xbmc/ContextMenuManager.cpp

```diff
@@ -230,21 +231,43 @@ bool CONTEXTMENU::ShowFor(const CFileItemPtr& fileItem, const CContextMenuItem&
   for (auto&& item : contextMenuManager.GetAddonItems(*fileItem, root))
     menuItems.emplace_back(std::move(item));
 
-  if (menuItems.empty())
+  CContextButtons buttons;
+  // compute fileitem property-based contextmenu items
+  {
+    int i = 0;
+    while (fileItem->HasProperty(StringUtils::Format("contextmenulabel({})", i)))
+    {
+      buttons.emplace_back(
+          ~buttons.size(),
+          fileItem->GetProperty(StringUtils::Format("contextmenulabel({})", i)).asString());
+      ++i;
+    }
+  }
```

Comment:
maybe this can be a private helper `CContextButtons GetPropertyMenuButtons(const CFileItem& item) const` or something.

Reply:
It's not easy because the context action is not part of the button, they're rather a simple label. As a result this is hard to encapsulate as later on you'll have to use the index to send the built-in message. I think it's better to try to reuse this method in media windows instead of duplicating the logic there